notebooks/Tutorial10_Geolocation_privacy_assessment.py

```diff
@@ -0,0 +1,251 @@
+
+# +
+selected_id = 1
+
+plt.figure(figsize=(10, 10))
+sns.scatterplot(
+    data=original_df[original_df["id"] == selected_id],
+    x="lon",
+    y="lat",
+    label="Original",
+)
+sns.scatterplot(
+    data=trimmed_df[trimmed_df["id"] == selected_id], x="lon", y="lat", label="Trimmed"
+)
+sns.scatterplot(
+    data=noised_df[noised_df["id"] == selected_id], x="lon", y="lat", label="Noised"
+)
+plt.title(
+    f"Example geolocation trace (id={selected_id}) and its trimmed and noised versions"
+)
+plt.legend()
+plt.show
+# -
+
+# ## Run a geolocation privacy assessment
+
+# ### Upload datasets
+
+original_dataset = client.pandas_integration.upload_dataframe(original_df)
+print(f"Dataset {original_dataset.id} loaded")
+trimmed_dataset = client.pandas_integration.upload_dataframe(trimmed_df)
+print(f"Dataset {trimmed_dataset.id} loaded")
+noised_dataset = client.pandas_integration.upload_dataframe(noised_df)
+print(f"Dataset {noised_dataset.id} loaded")
+
+# ### Create scenarios
+#
+# A privacy assessment for geolocation data requires the definition of potentially multiple scenarios.
+# A scenario defines some attacks that will be considered during the assessment. One scenario contains the following parameters:
+# - `projection_parameters` defines the method of projection (or representation) of the data that will be used to evaluate the level of protection against **singling-out attacks**. For example, an attack can attempt re-identification using trace density information (this projection is set by using `GeolocationDensityParameters`). An alternative is for re-identification of traces to be performed from calculated features such as mean speed, trip duration ... This projection can be chosen by setting the parameters with `GeolocationFeaturesParameters`.
+# - `known_features` defines a calculated feature that may be known to an attacker and that could be used to link the anonymized data with data from an external source. This attack is a **feature-based linkability attack**.
+# - `target_feature` defines a calculated feature that could be considered as sensitive and should be protected against **feature-based inference attacks**. Such attack would use the `known_features` to infer the `target_feature`.
+# - `known_signal_position` defines a part of the trip that could be known to an attacker (i.e. the start, middle or end of a trip). Attempt to re-associate anonymized trips to their original counterparts only using this information is considered as a **trace-based linkability attack**.
+# - `target_signal_poi` defines the point of interest (POI) that could be considered as a target for an attacker in the context of a **trace-based inference attack**. It can be the start or the end of a trip.
+# - `inference_metric_threshold` defines the distance under which a prediction on a target point of interest will be considered as a successful prediction by the attacker. Setting a large threshold will increase the expected level of protection.
```

Comment:
```suggestion
# - `inference_metric_threshold` defines the distance under which a prediction on a target point of interest will be considered as a successful prediction by the attacker. Setting a large threshold will decrease the expected level of protection.
```

For instance, you set the threshold at 100km. In almost all cases, the predicted POI will be within the 100km, and hence the attack will be considered successful -> protection rate equals 0%

Reply:
yes that's what I meant the way I wrote it, but I realise it can be confusing, I'll change it.